packages/web-components/src/calendar/calendar.template.ts

```diff
@@ -58,16 +95,20 @@ export function secondaryPanelCellTemplate(
   const cellTag = html.partial(tagFor(options.dataGridCell));
   return html`
       <${cellTag}
-          class="${(x, c) => c.parentContext.parent.getSecondaryCellClassNames(x.detail, todayMonth, todayYear)}"
+          class="${(x, c) => c.parentContext.parent.getSecondaryPanelCellClassNames(x.detail, todayMonth, todayYear)}"
+          id="id-secondary-panel-cell"
           part="secondary-panel-cell"
           tabindex="-1"
           role="gridcell"
           grid-column="${(x, c) => c.index + 1}"
+          aria-label="${(x, c) => c.parentContext.parent.getSecondaryPanelCellLabels(x.detail)}"
+          aria-selected="${(x, c) => c.parentContext.parent.getSecondaryPanelCellSelected(x.detail)}"
           @click="${(x, c) => c.parentContext.parent.$emit('secondaryPanelCellSelected', x.detail)}"
           @keydown="${(x, c) => c.parentContext.parent.handleSecondaryPanelKeydown(c.event as KeyboardEvent, x.detail)}"
       >
         <div
-        class="secondary-panel-cell">
+        class="secondary-panel-cell"
+        aria-labelledby="id-secondary-panel-cell">
```

Comment:
It looks like this is the intention of setting the `id`

If you decided to programmatically generate the secondary tab panels id then you can also programmatically set this.